### ISSUE TITLE
Unable to Deploy Zarf's Big Bang Core Package

### ISSUE THREAD

**Issue by erikschlegel** [defenseunicorns/zarf#610]
I'm unable to deploy zarf's big bang core package. When I ran the zarf deploy command I can see the big bang helm chart get installed (via the trace logs) but the flux source controller is unable to access the BB gitops repo from gitea. 

This is the error I see in the K8 console. 

_HelmChart 'bigbang/bigbang-bigbang' is not ready_

&& _no artifact available for GitRepository source 'bigbang'_

I can confirm that the gib bang gitops repo is available in gitea 

![image](https://user-images.githubusercontent.com/7635865/178110606-1155bce0-0bcb-4f62-b649-92674af32dac.png)

### Environment
Device and OS: Validated on both Azure AKS and K8 Kind Linux Ubuntu 20.04
App version: 0.19.6
Kubernetes distro being used: 1.21.9
Other: 

### Steps to reproduce
```
git clone https://github.com/defenseunicorns/zarf.git
cd zarf/packages/big-bang-core
zarf init --confirm --components git-server
zarf package create --confirm
zarf package deploy "zarf-package-big-bang-core-demo-amd64.tar.zst" -l=trace
```

### Expected result
I would expect that the big bang helm chart would be applied to the cluster and I can see the core namespaces provisioned with the appropriate resources.

### Actual Result
```
kubectl get ns
NAME                 STATUS   AGE
bigbang              Active   13h
default              Active   13h
flux-system          Active   13h
kube-node-lease      Active   13h
kube-public          Active   13h
kube-system          Active   13h
local-path-storage   Active   13h
zarf                 Active   13h
``` 

![image](https://user-images.githubusercontent.com/7635865/178110773-60f5dd05-f2c1-470a-8641-9937a721f573.png)

### Severity/Priority
😕 Blocked on deploying services to a big bang cluster via zarf

**Comment by jeff-mccoy** [comment]
There may be an issue with 19.6 (see #564), but the steps above are slightly confusing.  Can you confirm what commit or tag you are running the repo from?  Big Bang Core moved from examples to packages a couple versions ago.

**Comment by erikschlegel** [comment]
Apologies for the typo @jeff-mccoy. I’m running the deployment off main and meant to write the `packages` directory. I updated my repro steps in the issue description.

**Comment by jsburckhardt** [comment]
Hey folks, I'm having the same issue in k3d/aks. Basically, the gitrepository isn't getting created because the hook fails.
In the hook pod:
![image](https://user-images.githubusercontent.com/18494471/178273501-5afe8a9e-d8c1-46b8-814b-d829497bac98.png)
In the flux source:
![image](https://user-images.githubusercontent.com/18494471/178273715-0773fc51-858a-4563-ba37-df7f4ccd8254.png)

zarf version -> v0.19.6

reproduce issue:

init the cluster with v0.19.6 init binary
```bash
zarf init --components git-server --confirm 
```
create and deploy package
```bash
cd packages/big-bang-code
zarf package create --confirm
zarf package deploy 
(big-bang-code-standard option)
```

Hope that helps!
Cheers

**Comment by jeff-mccoy** [comment]
Thanks everyone, here's the summary for this issue and how we're addressing.  tl;dr;  If you want to try an immediate fix and you're on `v0.19.6`:

```sh
zarf package deploy sget://defenseunicorns/zarf-package-agent-hotfix:amd64
kubectl -n flux-system rollout restart deployment/source-controller
```

- A bug has existed for `Pod` and `GitRepository` resources on UPDATE since the webhook was added.  When we created the webhook, we only bound to CREATE operations and _not_ UPDATE operations.  For pods this wasn't important since deployments delete them on rollout.  However, Flux actually does UPDATE `GitRepository` resources.  

- v0.19.6 introduced a fix in #565 to resolve this, but we didn't account for the hook bindings in the webhook code that would also need to be updated for this to work.  There is some [discussion of idempotence](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#idempotence) in the K8s docs that we already abide by, so the change was trivial.  However, at the time we lacked testing for using the Flux part of the mutating webhook.  

- We quickly created a potential fix (same day), but [DI2ME](https://github.com/defenseunicorns/zarf-package-software-factory) was not blocked as they just used manual URls instead for now to avoid this issue.  We decided to try and finish a major test refactor (#581) that would prevent these types of issues in the future.  However since more users are encountering the issue now, we have also created the hotfix above until the next release is cut some time this week.
